File: Topico_4/uniao.hs
pertence n x = elem n (x)

uniao [] y = y
uniao (z:zs) y =
  if pertence z y
    then uniao zs y
    else z: uniao zs y

main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO [Int]
    print $ uniao a b

File: Topico_4/uniao.js
uniao = (list, list2) => {
  for(let i = 0; i < list2.length;i++){
    if(!list.find(n => n == list2[i])){
      list.push(list2[i]);
    }
  }
  return list;
}

console.log(uniao([1,2,3], [4,5,1]));